packages/fuels-accounts/src/supported_versions.rs

```diff
@@ -0,0 +1,116 @@
+use semver::Version;
+
+fn get_supported_fuel_core_version() -> Version {
+    "0.20.6".parse().unwrap()
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) struct VersionCompatibility {
+    pub(crate) supported_version: Version,
+    pub(crate) is_major_supported: bool,
+    pub(crate) is_minor_supported: bool,
+    pub(crate) is_patch_supported: bool,
+}
+
+pub(crate) fn check_fuel_core_version_compatibility(
+    network_version: &Version,
+) -> VersionCompatibility {
+    let supported_version = get_supported_fuel_core_version();
+    check_version_compatibility(network_version, &supported_version)
+}
+
+fn check_version_compatibility(
+    actual_version: &Version,
```

Comment:
If we accept owned values here, we can move the bellow call to clone() mora towards the edge of our API. I.e. we clone when passing arguments to
`check_fuel_core_version_compatibility`